lab.R

```diff
@@ -25,58 +25,78 @@ library(assertthat)
-#' 
-#' 
-#' 
+#' AwardedAmountToDate, StartDate, Oganization 
 
 
+#' 3. *Before diving into dates, let's do a quick manual check of the money column, `AwardedAmountToDate`.  The name of this column can be misleading; for most types of awards, it represents the total amount of the award, not just the amount that has been distributed to date.  We can view the information for a specific award using a URL like this: <https://www.nsf.gov/awardsearch/showAward?AWD_ID=1813043> (this should match the top row of your dataframe).  Award IDs are in the `AwardNumber` column. For the first 8 awards, compare the `AwardedAmountToDate` values in the dataframe to the "Awarded Amount to Date" values shown in the award search system.  Based on these 8 awards, does `AwardedAmountToDate` seem to contain valid and correctly-formatted data?* 
+ri_df_first8 <- ri_df %>% head(8)
+# Subset the dataset so I can look the two columns more closely
+part <- ri_df_first8[c("AwardNumber", "AwardedAmountToDate")]
+#' No. For Award IDs being 1652633 and 1651909, the "Awarded Amount to Date" values in the award search system and the "AwardedAmountToDate" values are different. 
 
 #' # Problem 2 #
 #' 
 #' 1. *The column `StartDate` is supposed to contain the date each program began.  What class is this column?  How are the dates formatted?  For convenience as we write some functions to parse these dates, create a variable `dates` that contains the dates from `StartDate`.* 
-#' 
-#' 
-#' 
+class(ri_df$StartDate)
+#' The class of the "StartDate" column is "character". 
+#' The dates are formatted differently. Some dates refer to a certain day, and some some days are the days passed since a reference date. 
+ri_df_dates <- ri_df %>% 
+  mutate(dates = StartDate)
 
 #' 2. *Let's start with the dates formatted like "43109."  These represent days passed since a reference date, which is sometimes called the "serial date" representation or format.  Typically — and in this case — the reference date is December 30, 1899 (for backwards compatibility with a bug in Lotus 1-2-3); but note that it may be December 31, 1903 on systems where Excel is trying to be compatible with an old Mac format. (More here: <http://www.cpearson.com/excel/datetime.htm>.)  The base R function `as.Date()` has an `origin` argument that can be used with integers to parse this date format:* 
 
 as.Date(2, origin = '1899-12-30')
 
 #' *Write a pipe that first uses `as.integer()` to convert the dates to integers, then passes them through `as.Date()`.  Assign the result to the variable `int_dates1` (1 for the "first attempt").*
-
+ri_df_dates <- ri_df_dates %>% 
+  mutate(int_dates1= dates) %>% 
+  mutate(int_dates1= as.integer(int_dates1)) %>% 
+  mutate(int_dates1 = as.Date(int_dates1, origin='1899-12-30'))
```

Comment:
Ahh I think I get where you're confused now.  I'm asking you to create these variables in the environment, not within the dataframe.  The idea is that it's easier to write a function that takes a single vector as input and apply it to a dataframe, rather than working directly with the whole dataframe.

Reply:
So for example, what I'm looking for here would look like
```
int_dates1 <- dates %>%
    as.integer() %>%
    as.Date(origin = '1899-12-30')
```

Reply:
I got it. Thanks!